packages/react/src/components/List/HierarchyList/HierarchyList.jsx

```diff
@@ -276,6 +286,33 @@ const HierarchyList = ({
     }
   }, [items, previousItems]);
 
+  /**
+   * Effect to change the currently expanded hierarchies. Ignore it on the first render to allow
+   * defaultExpandedIds to work, but fire it on all other changes
+   */
+  useEffect(() => {
+    if (!isFirstRender.current) {
+      setExpandedIds(expandedIdsProp);
+    }
+    isFirstRender.current = false;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [expandedIdsProp]);
+
+  /**
+   * effect to trigger calling the onExpandedChange callback. Ignore it on the initial render
+   * when previousExpandedIds is undefined, but fire it on every change afterward
+   */
+  useEffect(() => {
```

Comment:
Would it not be easier to use the `toggleExpansion` callback from the list, and also keep the APIs more aligned?

Reply:
I started to, but we call setExpandedIds in lots of other places (searches or when the items change, for example), so I thought the effect was cleaner than calling it in multiple places any time it changed.

Reply:
Ok, makes sense